rest_flex_fields/filter_backends.py

```diff
@@ -37,18 +37,17 @@ def filter_queryset(
         serializer.apply_flex_fields(serializer.fields, serializer._flex_options_rep_only)
         serializer._flex_fields_rep_applied = True
 
-        model_fields = [
-            self._get_field(field.source, queryset.model)
-            for field in serializer.fields.values()
-            if self._get_field(field.source, queryset.model)
-        ]
-
-        nested_model_fields = [
-            self._get_field(field.source, queryset.model)
-            for field in serializer.fields.values()
-            if self._get_field(field.source, queryset.model)
-            and field.field_name in serializer.expanded_fields
-        ]
+        model_fields = []
+        nested_model_fields = []
+        for field in serializer.fields.values():
+            model_field = self._get_field(field.source, queryset.model)
+            if model_field:
+                model_fields.append(model_field)
+                if (
+                    field.field_name in serializer.expanded_fields or
+                    (model_field.is_relation and not model_field.many_to_one)
```

Comment:
I think we should do it only for the fields listed in `expandable_fields`. And please move this condition to function like `_is_expandable_m2m` or like this.

Reply:
Hi Adrian!

Why do you think we should only include the fields listed in expandable_fields? **Even if they're not expanded, Django performs an extra query just to fetch the pks.**

I think we should include all m2m in prefetch_related present in the serializer fields (after aplying flex logic).

BR

Reply:
Hi! 
I think FlexFieldsFilterBackend should not be responsible for not "flex" fields. 
Because if "yes" then we have to `prefetch_related/select_related` regular (always expanded) FK field as well.
But we can just add an option for it. How do you think?

Also, please make sure we don't do unneeded queries. E. g. if the m2m field is expandable but not expanded.

## Example:
```
class UserSerializer(FlexFieldsModelSerializer):
    always_expanded_fk = ...
    always_expanded_m2m = ...
    never_expanded_m2m_ids = ...
    expandable_m2m_ids = ...

    expandable_fields = {
        ('expandable_m2m', {}),
        ('only_expandable_m2m', {}),
    }
```
### Current implementation:
Optimize:
- expandable_m2m
- only_expandable_m2m

### Your proposal:
Optimize:
- expandable_m2m
- only_expandable_m2m
- always_expanded_m2m
- never_expanded_m2m_ids
- expandable_m2m_ids

**Note: please, make sure we don't have extra queries for `expandable_m2m` and `only_expandable_m2m` when they are not expanded**

### My new proposal (not sure about it. Just an idea):
Optimize:
- expandable_m2m
- only_expandable_m2m

Optimize only if `auto_select_related_not_expandable=True`:
- always_expanded_fk
- always_expanded_m2m
- never_expanded_m2m_ids
- expandable_m2m_ids

Reply:
Hi @ADR-007;

I've just realized that this PR has been merged.

I read your reply the same day late in the night and I completely forgot to answer. I'm so sorry about that.

However, you can be confident that there are no unneeded queries in my proposal, as **Django always performs extra queries when dealing with m2m (expanded or not)**. The new code performs, at most, the same number of queries.

Back into your proposal, I can't imagine a scenario in which a user could prefer NOT to optimize the queryset, so I would always optimize every field in the serializer. Remember that you're working with **the flexible version of the serializer, once all unwanted fields have been "removed"**. In my case, I would always set `auto_select_related_not_expandable=True`.

Feel free to revert/improve the changes I made if you're not comfortable with the new code/solution. As I said before, I forgot this PR was still open.

Best regards
Happy 2022!